doc/psud/PSU_daemon_design.md

```diff
@@ -168,14 +270,52 @@ class PsuBase(device_base.DeviceBase):
 	
     def get_input_current(self):
         raise NotImplementedError
-    ...
 
+    def get_psu_power_warning_suppress_threshold(self)
+        """
+        The value can be volatile, so the caller should call the API each time it is used.
+
+        Returns:
+            A float number, the warning threshold of the PSU in watts.
+        """
+        raise NotImplementedError
+
+    def get_psu_power_critical_threshold(self)
+        """
+        Retrieve the critical threshold of the power on this PSU
+        The value can be volatile, so the caller should call the API each time it is used.
+
+        Returns:
+            A float number, the critical threshold of the PSU in watts.
+        """
+        raise NotImplementedError
 ```
 
 ### 6. PSU daemon flow
 
 Supervisord takes charge of this daemon. This daemon will loop every 3 seconds and get the data from psuutil/platform API and then write it the Redis DB.
 
-- The psu_num will store in "chassis_info" table. It will just be invoked one time when system boot up or reload. The key is chassis_name, the field is "psu_num" and the value is from get_psu_num(). 
+- The psu_num will store in "chassis_info" table. It will just be invoked one time when system boot up or reload. The key is chassis_name, the field is "psu_num" and the value is from get_psu_num().
 - The psu_status and psu_presence will store in "psu_info" table. It will be updated every 3 seconds. The key is psu_name, the field is "presence" and "status", the value is from get_psu_presence() and get_psu_num().
 - The daemon query PSU event every 3 seconds via platform API. If any event detects, it should set PSU LED color accordingly and trigger proper syslog.
+
+### 7. Test cases
+
+#### 7.1 Unit test cases added for PSU power exceeding checking
+
+1. Neither `get_psu_power_warning_suppress_threshold` nor `get_psu_power_critical_threshold` is supported by platform API when a new PSU is identified
+   In `psu_status`, power exceeding check should be stored as `not supported` and no further function call.
+2. Both `get_psu_power_warning_suppress_threshold` and `get_psu_power_critical_threshold` are supported by platform API when a new PSU is identified
+   In `psu_status`, power exceeding check should be stored as `supported`
+3. PSU's power was less than the warning threshold and is in the range (warning threshold, critical threshold): no action
+4. PSU's power was in range (warning threshold, critical threshold) and is greater than the critical threshold
+   1. if warning was raised, no action expected
+   2. if warning was not raised, a warning should be raised
+5. PSU's power was less than the warning threshold and is greater than the critical threshold: a warning should be raised
```

Comment:
its confusing

Reply:
It's to test the case that the PSU power jumps from a value below the warning suppress threshold to a value above critical threshold.